app/v1/models/domain/BFLoss.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.models.domain
+
+import play.api.libs.json._
+import v1.models.des.DesBFLoss
+import v1.models.requestData.DesTaxYear
+
+case class BFLoss(typeOfLoss: String,
+                  selfEmploymentId: Option[String],
+                  taxYear: String,
+                  lossAmount: BigDecimal) {
+
+  def toDes(broughtForwardLoss: BFLoss) : DesBFLoss = {
+    DesBFLoss(lossType = typeOfLoss match {
```

Comment:
What do we want to happen if it is not one of these four types? Or are we going to leave it up to the request parser to catch this?

Reply:
I assume the parser/validator?

Reply:
I have written code that will handle this.